### PR TITLE
TBDGen: Specify the correct macCatalyst platform ID in `$ld$previous` directives

### PR DESCRIPTION
**Pull request by tshortli**
Resolves rdar://123491072

### CODE REVIEW

**Comment by tshortli** [comment]
@swift-ci please test

File: test/TBD/linker-directives-ld-previous-macos.swift [hunk 1]
```diff
@@ -2,24 +2,27 @@
...
-// CHECK: D $ld$previous$/System/Previous/macOS/ToasterKit.dylib$$1$10.8$10.15$_$s10ToasterKit7VehicleVMa$
-// CHECK: D $ld$previous$/System/Previous/macOS/ToasterKit.dylib$$1$10.8$10.15$_$s10ToasterKit7VehicleVMn$
-// CHECK: D $ld$previous$/System/Previous/macOS/ToasterKit.dylib$$1$10.8$10.15$_$s10ToasterKit7VehicleVN$
+// RUN: %target-swift-frontend -target x86_64-apple-ios13.1-macabi -typecheck %S/Inputs/linker-directive.swift -emit-tbd -emit-tbd-path %t/linker_directives_macabi.tbd -previous-module-installname-map-file %S/Inputs/install-name-map-toasterkit.json -tbd-install_name toasterkit
+// R/UN: %llvm-nm %t/linker_directives_macabi.tbd | %FileCheck -check-prefixes=CHECK,CHECK-MACCATALYST %s --implicit-check-not "System/Previous/macOS"
 
-// CHECK-ZIPPERED: $ld$previous$/System/Previous/iOS/ToasterKit.dylib$$2$10.2$13.0$_$s10ToasterKit5toastyyF$
-// CHECK-ZIPPERED: $ld$previous$/System/Previous/macOS/ToasterKit.dylib$$1$10.8$10.15$_$s10ToasterKit5toastyyF$
+// CHECK-MACCATALYST: D $ld$previous$/System/Previous/macCatalyst/ToasterKit.dylib$$6$10.2$13.0$_$s10ToasterKit5toastyyF$
```
Comment (ian-twilightcoder) [inline test/TBD/linker-directives-ld-previous-macos.swift:16]: I'm pretty sure the versions are wrong here and they need to be the iOS versions mapped from the SDKSettings. We should also test that 10.15/13.1 is the earliest version. i.e. something available in 10.14 and moved in 12.0 should make an $ld$previous$ of 13.1$15.0. And something available in 10.14 and moved in 10.15 shouldn't make an $ld$previous$ for Mac Catalyst at all.

**Review by ian-twilightcoder — CHANGES_REQUESTED** [review]

File: test/TBD/linker-directives-ld-previous-macos.swift (hunk 1 above)
Comment (tshortli) [reply test/TBD/linker-directives-ld-previous-macos.swift:16]: `10.2` and `13` come directly from the _iOS_ availability of the declaration (see https://github.com/apple/swift/blob/main/test/TBD/Inputs/linker-directive.swift). So I don't think a remapping is needed. I can see about flooring the versions at 13.1 though.

Comment (tshortli) [reply test/TBD/linker-directives-ld-previous-macos.swift:16]: Actually, I'd like to separate clamping the values to minimum OS versions and potentially skipping emission of directives out into a separate radar/PR. I agree it's the right thing to do but it's also applicable to all platforms and it's unclear to me whether the current behavior is concretely causing any issues. I think it should be treated as a separate bug fix from a scheduling and risk perspective.

Comment (ian-twilightcoder) [reply test/TBD/linker-directives-ld-previous-macos.swift:16]: Oh you're right, I saw "10.something" and got mixed up.

**Review by nkcsgexi — APPROVED** [review]
Thank you for fixing it, Allan!

**Review by ian-twilightcoder — APPROVED** [review]

**Pull request merged**